lib/Module/Optimize.cpp

```diff
@@ -183,7 +196,18 @@ void Optimize(Module *M, llvm::ArrayRef<const char *> preservedFunctions) {
     // for a main function.  If main is defined, mark all other functions
     // internal.
     if (!DisableInternalize) {
+#if LLVM_VERSION_CODE >= LLVM_VERSION(3, 9)
+      auto PreserveFunctions = [=](const GlobalValue &GV) {
+	StringRef GVName = GV.getName();
+	for (const char *fun: preservedFunctions)
+	  if (GVName.equals(fun))
+	    return true;
+	  return false;
+      };
```

Comment:
Another minor thing: please use spaces only (instead of mixture with tabs) in the lambda's body.

Reply:
...and fix the indentation of the last line (which is not part of the for-loop).

Reply:
Fixed.